compiler/front/cli_reporter.nim

```diff
@@ -1600,7 +1600,16 @@ proc reportBody*(conf: ConfigRef, r: SemReport): string =
         "` can be defined only in the same module with its type (" & r.typ.render & ")"
 
     of rsemUnexpectedTypeBoundOpSignature:
-      result = "signature for '" & r.symstr & "' must be proc[T: object](x: var T)"
+      let value =
+        case r.symstr
+        of "=copy":    "proc(x: var T, y: T)"
+        of "=destroy": "proc(x: var T)"
+        of "=sink":    "proc(x: var T, y: T)"
+        of "=trace":   "proc(x: var T, env: pointer)"
+        else:          unreachable()
```

Comment:
We don't align of branches in skull (or nim, for that matter).

You should add a message to `unreachable()`, like `unreachable("unexpected hook: " & r.symstr)`. This would allow a future developer to quickly figure out what case to add.

I'm not sure myself if `unreachable()` is the tool to use here, or if we should go with an `assert` instead to only error in development builds. That would prevent a missed case from crashing the compiler. A fallback string like `FIXME: unknown hook signature` should do.

Reply:
You should add a case for [`deepcopy`](https://github.com/nim-works/nimskull/blob/fb920984a449b72d45b78e227048412f8ca97a66/doc/manual_experimental.rst?plain=1#L1063-L1081) as well.

Reply:
> We don't align of branches in skull (or nim, for that matter).

I'm personally ok with it in this case as it does make it easier to read